### PR TITLE
Issue 444 will start update issue

### PR DESCRIPTION
**Pull request by aab-odoo**

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ export function useStore(selector, options: SelectorOptions = {}): any {
     updateFunctions.forEach(function(updateFn) {
```
Comment (ged-odoo) [inline src/store.ts]: we should use a for loop here. foreach + function is much slower, and this is a hot code path

**Comment by ged-odoo** [comment]
this pr looks good to me...

**Comment by aab-odoo** [comment]
@ged-odoo I added a test case for the scenario we were talking about. It passes, thanks to [this](https://github.com/odoo/owl/blob/master/src/context.ts#L149) making [this test](https://github.com/odoo/owl/blob/master/src/context.ts#L142) always false. So if we can guarantee that onWillUnmount is called before the update callback, it's fine.

**Comment by ged-odoo** [comment]
good. the test will guarantee that we keep this invariant in the future.

**Pull request merged**